### ISSUE TITLE
Ideas of LLM Providers to Implement in gollm

### ISSUE THREAD

**Issue by teilomillet** [teilomillet/gollm#4]
We're always looking to expand gollm's capabilities by adding support for more LLM providers. 

If you're interested in contributing, here's a list of providers we'd love to see integrated:

- [ ] Mistral
- [ ] Cohere
- [ ] OpenRouter
- [ ] Lambda Labs
- [ ] Google (VertexAI)
- [ ] AWS Bedrock
- [ ] Microsoft Azure

This list is not exhaustive or in any particular order. If you're interested in working on one of these providers or have another provider in mind:

- Check our CONTRIBUTING.md for guidelines on how to contribute.
- Feel free to ask any questions if you need help or clarification.

We welcome suggestions for other providers not on this list. If you think another provider would be a valuable addition to gollm, please let us know!

We'll update this list as new suggestions come in or as providers are successfully integrated. Keep an eye on this issue for the latest opportunities to contribute!

Thank you !

**Comment by go-dockly** [comment]
https://huggingface.co/chat/
https://www.orbitfin.ai/
https://finchat.io/

**Comment by lorenzomainardi** [comment]
What about Ollama? Do you support it?

**Comment by teilomillet** [comment]
@lorenzomainardi yes Ollama is supported, please check the example to know more about it : https://github.com/teilomillet/gollm/blob/main/examples/ollama_example.go

or the code if you know more about it :
https://github.com/teilomillet/gollm/blob/main/providers/ollama.go

**Comment by andrejsstepanovs** [comment]
Is there a way to inject my own provider and use it?

**Comment by teilomillet** [comment]
## Provider Updates

The following providers have been added or updated:

### Newly Added Providers
- **Lambda Labs** (`lambda`) - OpenAI-compatible API for Lambda Labs' cloud GPU models
- **AWS Bedrock** (`bedrock`) - Support for models via AWS Bedrock (Claude, Llama, Mistral, Cohere, etc.) with AWS SigV4 authentication

### Updated Providers  
- **OpenRouter** (`openrouter`) - Now properly registered in the default provider registry

### Example Usage

```go
// Lambda Labs
llm, _ := gollm.NewLLM(
    gollm.SetProvider("lambda"),
    gollm.SetAPIKey(os.Getenv("LAMBDA_API_KEY")),
    gollm.SetModel("hermes-3-llama-3.1-405b-fp8"),
)

// AWS Bedrock (credentials from AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY)
llm, _ := gollm.NewLLM(
    gollm.SetProvider("bedrock"),
    gollm.SetAPIKey(""), // Uses AWS env vars
    gollm.SetModel("anthropic.claude-3-sonnet-20240229-v1:0"),
)
```

See `examples/lambda/` and `examples/bedrock/` for more complete examples.

### Updated Checklist
- [x] Mistral
- [x] Cohere
- [x] OpenRouter
- [x] Lambda Labs ✨ NEW
- [x] AWS Bedrock ✨ NEW
- [ ] Google VertexAI (planned - requires Google Cloud SDK)
- [x] Microsoft Azure (as azure-openai)
